src/swtpm_setup/swtpm.c

```diff
@@ -128,8 +129,19 @@ static int swtpm_start(struct swtpm *self)
     }
 #endif
 
-    success = g_spawn_async(NULL, argv, NULL,
-                            G_SPAWN_LEAVE_DESCRIPTORS_OPEN | G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL,
+    flags = G_SPAWN_LEAVE_DESCRIPTORS_OPEN;
+    if (gl_LOGFILE) {
+        flags |= G_SPAWN_STDOUT_TO_DEV_NULL | G_SPAWN_STDERR_TO_DEV_NULL;
+    } else {
+#ifdef G_SPAWN_CHILD_INHERITS_STDOUT
+        flags |= G_SPAWN_CHILD_INHERITS_STDOUT;
```

Comment:
The flags are typedefs and not #defines...  it won't set the flags then at all unless I am missing something. I suppose you did this to accommodate possibly older versions of glib-2.0 that don't have these flags?
```
typedef enum
{
  G_SPAWN_DEFAULT                = 0,
  G_SPAWN_LEAVE_DESCRIPTORS_OPEN = 1 << 0,
  G_SPAWN_DO_NOT_REAP_CHILD      = 1 << 1,
  /* look for argv[0] in the path i.e. use execvp() */
  G_SPAWN_SEARCH_PATH            = 1 << 2,
  /* Dump output to /dev/null */
  G_SPAWN_STDOUT_TO_DEV_NULL     = 1 << 3,
  G_SPAWN_STDERR_TO_DEV_NULL     = 1 << 4,
  G_SPAWN_CHILD_INHERITS_STDIN   = 1 << 5,
  G_SPAWN_FILE_AND_ARGV_ZERO     = 1 << 6,
  G_SPAWN_SEARCH_PATH_FROM_ENVP  = 1 << 7,
  G_SPAWN_CLOEXEC_PIPES          = 1 << 8
} GSpawnFlags;
```

Reply:
It looks like `G_SPAWN_CHILD_INHERITS_STDOUT/STDERR` were introduced in 2.74: https://github.com/GNOME/glib/blob/bfe349983a7f32d969485cd29d8ce17e06121284/glib/gspawn.h#L191-L207

You may need to use the GLIB_CHECK_VERSION() macro in place of the two #ifdef's. (Example may be here: https://thrift.apache.org/tutorial/c_glib.html)

Reply:
Oh, I missed that it was a #define. Will fix.